src/BloomExe/Edit/EditingModel.cs

```diff
@@ -1245,9 +1257,10 @@ private static string MassageUrlForJavascript(string url)
 			return newUrl.Replace(':', '$').Replace('\\', '/');
 		}
 
-		internal void ChangePageLayout()
+		internal void ChangePageLayout(IPage page)
 		{
-			_view.ShowChangeLayoutDialog();
+			PageChangingLayout = page;
```

Comment:
I put in some code to handle it being null, but as things currently work, it is always set by the code that launches the dialog.